src/mat73.c

```diff
@@ -2200,11 +2200,15 @@ Mat_Create73(const char *matname,const char *hdr_str)
     mat_t *mat = NULL;
     size_t err;
     time_t t;
-    hid_t plist_id,fid;
+    hid_t plist_id,fid,plist_ap;
 
     plist_id = H5Pcreate(H5P_FILE_CREATE);
     H5Pset_userblock(plist_id,512);
-    fid = H5Fcreate(matname,H5F_ACC_TRUNC,plist_id,H5P_DEFAULT);
+    plist_ap = H5Pcreate(H5P_FILE_ACCESS);
```

Comment:
plist_ap is not closed in case of the two early returns in line 2217 and 2224.

Reply:
Whats preferable - closing in the error path, or closing always and recreating the FAPL prior to H5Fopen?

Reply:
Hm, I have no strong opinion here, but closing in the error path sounds ok.